smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/ProtocolGenerator.kt

```diff
@@ -86,6 +88,9 @@ interface ProtocolGenerator {
     val unknownServiceErrorSymbol: Symbol
         get() = DefaultUnknownServiceErrorSymbol
 
+    val retryErrorInfoProviderSymbol: Symbol
+        get() = DefaultRetryErrorInfoProviderSymbol
```

Comment:
Provides the correct type for the retry error info provider.

The AWS SDK overrides this with its custom type.